lib/src/conflicts.rs

```diff
@@ -62,47 +62,28 @@ impl<T> Conflict<T> {
     }
 }
 
-impl Conflict<Option<TreeValue>> {
-    /// Create a `Conflict` from a `backend::Conflict`, padding with `None` to
-    /// make sure that there is exactly one more `adds()` than `removes()`.
-    pub fn from_backend_conflict(conflict: &backend::Conflict) -> Self {
-        let mut removes = conflict
-            .removes
-            .iter()
-            .map(|term| Some(term.value.clone()))
-            .collect_vec();
-        let mut adds = conflict
-            .adds
-            .iter()
-            .map(|term| Some(term.value.clone()))
-            .collect_vec();
+impl<T: Clone> Conflict<T> {
+    /// Create a `Conflict` from a `removes` and `adds`, padding with `None` to
+    /// make sure that there is exactly one more `adds` than `removes`.
+    pub fn from_legacy_form(removes: Vec<T>, adds: Vec<T>) -> Conflict<Option<T>> {
+        let mut removes = removes.into_iter().map(Some).collect_vec();
+        let mut adds = adds.into_iter().map(Some).collect_vec();
         let num_diffs = max(removes.len() + 1, adds.len()) - 1;
         removes.resize(num_diffs, None);
         adds.resize(num_diffs + 1, None);
         Conflict { removes, adds }
     }
+}
 
-    /// Creates a `backend::Conflict` from a `Conflict` by dropping `None`
-    /// values. Note that the conflict is lossy: the order of `None` values is
-    /// not preserved when converting back to a `Conflict`.
-    pub fn to_backend_conflict(&self) -> backend::Conflict {
-        let removes = self
-            .removes
-            .iter()
-            .flatten()
-            .map(|value| backend::ConflictTerm {
-                value: value.clone(),
-            })
-            .collect_vec();
-        let adds = self
-            .adds
-            .iter()
-            .flatten()
-            .map(|value| backend::ConflictTerm {
-                value: value.clone(),
-            })
-            .collect_vec();
-        backend::Conflict { removes, adds }
+impl<T: Clone> Conflict<Option<T>> {
+    /// Creates lists of `removes` and `adds` from a `Conflict` by dropping
+    /// `None` values. Note that the conversion is lossy: the order of `None`
+    /// values is not preserved when converting back to a `Conflict`.
+    pub fn to_legacy_form(&self) -> (Vec<T>, Vec<T>) {
```

Comment:
As I was writing the chmod PR, I missed a non-allocating destructuring function
`Conflict<T>->(Vec<T>, Vec<T>)`.

Reply:
Ah, I think I also wanted that at some point. Done. (Well, it's a non-cloning function, but it still allocates vectors.)

Reply:
Relatedly, I'm hoping this could maybe simplify the "update conflicts from file" function and make function for addressing `i`-th add/remove less necessary. I might be missing something, though.

Reply:
Ah, I think I also wanted that at some point. Done. (Well, it's a non-cloning function, but it still allocates vectors.)

Reply:
I think this is good as far as this PR goes, but just to be clear, I meant a function `Conflict<T> -> (Vec<T>, Vec<T>)`. I might try creating that function and seeing whether it simplifies chmod and update_conflicts_from_file in a separate PR.

Reply:
I see. Makes sense to also have that function.

Reply:
I can't actually find a way to easily simplify `update_conflicts_from_file` with such a function. OTOH, the functions `set_add` and `set_remove` that I'd want to replace with such a function currently have no uses.